libraries/Parsers/Microsoft.Bot.Builder.Parsers.LU/Parser/Diagnostic.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+#pragma warning disable CS1591 // Missing XML comment for publicly visible type or member
+using Antlr4.Runtime;
+
+namespace Microsoft.Bot.Builder.Parsers.LU.Parser
+{
+    public enum DiagnosticSeverity
+    {
+        /// <summary>Raise an error</summary>
+        Error,
+
+        /// <summary>Raise warning instead of error</summary>
+        Warn
+    }
+
+    public class Diagnostic : Error
+    {
+        public Diagnostic(Range range, string message, string severity = "ERROR")
+        {
+            Range = range;
+            Message = message;
+            Severity = severity;
+        }
+
+        public static Diagnostic BuildDiagnostic(string message, DiagnosticSeverity severity = DiagnosticSeverity.Error, Range range = null, ParserRuleContext context = null)
+        {
```

Comment:
Can mesage be empty?
```suggestion
        if (string.IsNullOrEmpty(message))
        {
            throw new ArgumentException(nameof(message));
        }
```

Reply:
Yes